include/lsst/afw/geom/Angle.h

```diff
@@ -3,241 +3,294 @@
+     *
+     * @exceptsafe Shall not throw exceptions.
+     *
+     * @warning Exact limits are only guaranteed for radians; limits for other
+     * units may be slightly squishy, due to roundoff errors. Whether there are
+     * any violations is unknown; please update this comment if you can prove
+     * that the limits are or are not valid for all supported units.
+     */
+    Angle wrapCtr() const noexcept;
+
+    /**
+     * Wrap this angle to a value `x` such that -&pi; &le; `x - refAng` < &pi;.
+     *
+     * @param refAng reference angle to match
+     *
+     * @returns an angle in the custom normalized interval.
+     *
+     * @exceptsafe Shall not throw exceptions.
+     *
+     * @warning Exact limits are only guaranteed for radians; limits for other
+     * units may be slightly squishy due to roundoff errors. There are known
+     * violations that are demonstrated in testWrap in tests/angle.py.
+     */
+    Angle wrapNear(Angle const& refAng) const noexcept;
+
+    /**
+     * The signed difference between two Angles.
+     *
+     * @param other the angle to which this angle will be compared
+     * @return `*this - other`, wrapped to the range [-&pi;, &pi;)
+     *
+     * @exceptsafe Shall not throw exceptions.
+     */
+    Angle separation(Angle const& other) const noexcept;
+
+#define ANGLE_OPUP_TYPE(OP, TYPE)                \
+    Angle& operator OP(TYPE const& d) noexcept { \
+        _val OP d;                               \
+        return *this;                            \
     }
 
-ANGLE_OPUP_TYPE(*=, double)
-ANGLE_OPUP_TYPE(*=, int)
-ANGLE_OPUP_TYPE(+=, double)
-ANGLE_OPUP_TYPE(+=, int)
-ANGLE_OPUP_TYPE(-=, double)
-ANGLE_OPUP_TYPE(-=, int)
+    //@{
+    /// Multiply this angle by the given factor.
+    ANGLE_OPUP_TYPE(*=, double)
+    ANGLE_OPUP_TYPE(*=, int)
+    //@}
+    //@{
+    /// Increase this angle by the given number of radians.
+    ANGLE_OPUP_TYPE(+=, double)
+    ANGLE_OPUP_TYPE(+=, int)
+    //@}
+    //@{
+    /// Decrease this angle by the given number of radians.
+    ANGLE_OPUP_TYPE(-=, double)
+    ANGLE_OPUP_TYPE(-=, int)
+//@}
 
 #undef ANGLE_OPUP_TYPE
 
-#define ANGLE_COMP(OP)                          \
-    bool operator OP ( const Angle& rhs ) {     \
-        return _val OP rhs._val;                \
-    }
-
-ANGLE_COMP(==)
-ANGLE_COMP(!=)
-ANGLE_COMP(<=)
-ANGLE_COMP(>=)
-ANGLE_COMP(<)
-ANGLE_COMP(>)
+#define ANGLE_COMP(OP) \
+    constexpr bool operator OP(const Angle& rhs) const noexcept { return _val OP rhs._val; }
+
+    //@{
+    /// Test if two Angles represent the same angle (without wrapping).
+    ANGLE_COMP(==)
+    ANGLE_COMP(!=)
+    //@}
+    //@{
+    /// Compare the sizes of two Angles (without wrapping).
+    ANGLE_COMP(<=)
+    ANGLE_COMP(>=)
+    ANGLE_COMP(<)
+    ANGLE_COMP(>)
+//@}
 
 #undef ANGLE_COMP
 
 private:
     double _val;
 };
 
-Angle const NullAngle = Angle(-1000000., degrees);
-
+Angle constexpr NullAngle = Angle(-1000000., degrees);
```

Comment:
Please get rid of this. A global search turned up no usage of this in the stack and we use `nan` for an unknown angle.